internal/config/config_clientapi.go

```diff
@@ -12,24 +12,25 @@ type ClientAPI struct {
 	Listen Address `yaml:"listen"`
 	Bind   Address `yaml:"bind"`
 
+	// If set disables new users from registering (except via shared
+	// secrets)
+	RegistrationDisabled bool `yaml:"registration_disabled"`
 	// If set, allows registration by anyone who also has the shared
 	// secret, even if registration is otherwise disabled.
 	RegistrationSharedSecret string `yaml:"registration_shared_secret"`
+
+	// Boolean stating whether catpcha registration is enabled
+	// and required
+	RecaptchaEnabled bool `yaml:"enable_registration_captcha"`
```

Comment:
So this is exactly what I was trying to get at before when I mentioned about autogenning a config file from go source code. At the moment we have two descriptions of fields, this one and the one in `dendrite-config.yaml`. I would prefer it if:
 - Source of truth is the code comment above.
 - `generate-config` parses this file and writes out a YAML file with these comments.
 - We can have `Default: foo` and parse that to set the default when generating a config.
 - We check in the generated config.

WDYT?

Reply:
Broadly I'm not against this at all, as it means that our code annotates the config itself, but that's probably a longer-term activity than something for this PR.

Reply:
Yep just running the idea past you. I've done this thing before so I can tackle it.